### ISSUE TITLE
Builds but cannot run...

### ISSUE THREAD

**Issue by kodpraktikanten** [estatio/estatio#6]
Hi.

I have built a fresh clone from the master and it builds but it will not run correctly when I launch it as I did in my previous learning sessions. I am using the following command to run the self-hosted variant built using the "jetty-console" profile:
```
> java -jar estatioapp\webapp\target\estatio-webapp-1.2.0-SNAPSHOT-jetty-console.war
```
The problem I get are these type of exceptions and the embedded web server will not even start:
```
Exception in thread "Thread-4" java.lang.VerifyError: Bad type on operand stack
Exception Details:
  Location:
    org/estatio/dom/lease/invoicing/InvoiceItemForLease.compareTo(Lorg/estatio/dom/invoice/InvoiceItem;)I @2: invokespecial
  Reason:
    Type 'org/estatio/dom/invoice/InvoiceItem' (current frame, stack[1]) is not assignable to 'org/estatio/dom/UdoDomainObject'
  Current Frame:
    bci: @2
    flags: { }
    locals: { 'org/estatio/dom/lease/invoicing/InvoiceItemForLease', 'org/estatio/dom/invoice/InvoiceItem' }
    stack: { 'org/estatio/dom/lease/invoicing/InvoiceItemForLease', 'org/estatio/dom/invoice/InvoiceItem' }
  Bytecode:
    0000000: 2a2b b700 c03d 1c99 0005 1cac 2bc1 0002
    0000010: 9900 0fb2 00c2 2a2b c000 02b6 00c8 ac2a
    0000020: b600 ceb6 00d4 2bb6 00ce b600 d4b6 00d9
    0000030: ac
  Stackmap Table:
    append_frame(@12,Integer)
    same_frame(@31)

        at java.lang.Class.getDeclaredMethods0(Native Method)
        at java.lang.Class.privateGetDeclaredMethods(Class.java:2570)
        at java.lang.Class.privateGetPublicMethods(Class.java:2690)
        at java.lang.Class.getMethods(Class.java:1467)
        at org.apache.isis.core.metamodel.specloader.specimpl.FacetedMethodsBuilder.<init>(FacetedMethodsBuilder.java:145)
        at org.apache.isis.core.metamodel.specloader.specimpl.dflt.ObjectSpecificationDefault.<init>(ObjectSpecificationDefault.java:100)
```
Regards

**Comment by danhaywood** [comment]
I'm guessing you are running on Java 8?  Can you try Java 7.

**Comment by kodpraktikanten** [comment]
Hi.

I am running Java 7 on my learning machine where I have tested this.

Regards

**Comment by kodpraktikanten** [comment]
Hmm, seems as if some update also has placed Java 8 on my machine. I will remove it and test again.

Regards

**Comment by kodpraktikanten** [comment]
Nope, it does not work even when I have removed the Java 8 JRE just to be sure. All my environment settings points to the Java 7 installation.

I have also tested an old clone of the master branch from before Xmas and that one builds and executes.

Regards

**Comment by danhaywood** [comment]
Um, given that you did have Java 8 (and taking [1] as a bit of a clue,
where it says: "In order to avoid the VerifyError, you must compile all
your classes using the same version of Java"), then perhaps you could clear
out ~/.m2/repository/org/estatio and then build everything again, from the
root pom.xml.

My guess is that the stuff we factored out into udo-base was built with
Java8, and that might be the cause of the issue.

Dan


[1]
http://examples.javacodegeeks.com/java-basics/exceptions/java-lang-verifyerror-how-to-solve-verifyerror/

**Comment by kodpraktikanten** [comment]
Hi again.

You will have to excuse me but I am still having this problem. I have cleared my maven repositories and rebuilt everything using Java 7. Still the same result when I try to execute Estatio. I can for example execute the ToDo app built within the Estatio Isis fork.

- Is Java 7 preferred for Estatio or should I install and use Java 8 instead?

BTW, currently I cannot include the tests when building Estatio since I get an error in the tests.

Regards

**Comment by martin-g** [comment]
The problem with the broken test is the upgrade of Reflections in Isis to 0.9.9.
Updating Estatio's pom.xml and two minor API changes fixes the build here.

**Comment by danhaywood** [comment]
There were also some additional fixes [1] in Isis 1.8.0-SNAPSHOT itself.
These should now be in the snapshot repo, but you might want to delete out
your local ~/.m2/repository/org/apache/isis to force a download of the
latest-n-greatest.

Dan




[1]
https://github.com/apache/isis/commit/f65d37da39b4b6838a9e76e490b7a20184dc07e4

**Comment by kodpraktikanten** [comment]
Hi.

As I wrote yesterday I still cannot run Estatio. It is exact the same problem as I mentioned in my first post in this issue thread. I have now also done a complete test on another developer machine that I totally cleaned before the tests and the same problem appears there. I am running on Windows 7 with Oracle JDK 1.7 Update 76. 

- Is there anything special I need to pay attention to when setting up the build and execution environment for Estatio?
- Is the instruction about building Isis from the Estatio copy still correct or is Isis now used directly from the original master?

Regards

**Comment by martin-g** [comment]
I use JDK 1.7 update 72 on Ubuntu and Isis from original master. No problems here.

**Comment by kodpraktikanten** [comment]
Hi again.

I have now also done another test where I just deployed the Estatio war-file to a standalone Tomcat web server and then I can run Estatio. It seems to be a problem that appears when running the embedded Jetty console as I did previously on Windows 7 and Oracle JDK 1.7 Update 76. 

The interesting thing is that I can run the Isis demo apps (for example the ToDo app) using the Jetty console but not Estatio. Perhaps there is some class loader problem occuring in Estatio due to the packaging structure of classes in different jar-files?

Regards
